src/server/utils.js

```diff
@@ -0,0 +1,51 @@
+import { Access as AccessCapabilities, Space as SpaceCapabilities } from '@web3-storage/capabilities'
+import { InvalidDelegation } from '../middleware/withDelegationsStorage.js'
+
+/**
+ * Checks if the space/content/serve/* delegation is for the gateway and it is not expired.
+ *
+ * @param {import('@ucanto/interface').Signer} gatewayIdentity - The signer of the gateway identity
+ * @param {import('@ucanto/interface').InferInvokedCapability<typeof AccessCapabilities.delegate>} capability - The capability to validate
+ * @param {import('@ucanto/interface').Proof[]} proofs - The proofs to validate
+ */
+export const extractContentServeDelegation = (gatewayIdentity, capability, proofs) => {
+  const nbDelegations = new Set(Object.values(capability.nb.delegations))
+  if (nbDelegations.size !== 1) {
+    return { error: new InvalidDelegation(`nb.delegations has more than one delegation`) }
```

Comment:
It was originally designed to handle only a single delegation, specifically `space/content/serve/*`. However, as we discussed, this check is no longer necessary.

Reply:
After giving it more thought, I’ve decided to leave it as is for now since I won’t be implementing the verification for `transport/http` and `egress/record` at this time. Since these can be derived from the top-level capability, we can adjust the code later if we decide to implement individual delegations for specific sub-capabilities.